hexgame/src/board.py

```diff
@@ -67,30 +67,94 @@ def place_stone(self, i:int, j:int, color : Color) -> 'Board':
     """
 
     def has_cell(self, coords: tuple[int, int]) -> bool:
-        return False
+        x,y = coords
+        return (0 <= x < self.dim_x) and (0 <= y < self.dim_y)
+
+
+
+    """
+    is_border_cell checks if the @param cell is found on one of the 4 borders
+    of the board
+    """
+    def is_border_cell(self ,coords: tuple[int, int]) -> bool:
+        x,y = coords
+        return x == 0 or x == (self.dim_x -1) or y == 0 or y == (self.dim_y -1)
+
 
     """
     find_neighbours function finds all neighbouring cells
     in the board to the cell defined by @param coords
     @return list of neighbouring cells
     """
 
-    def find_neighbours(self, coords: tuple[int, int]) -> set[Cell]:
-        return []
+
+    def find_neighbours(self, coords: tuple[int,int]) -> set[Cell]:
+        #TODO: finish this
+        x,y = coords
+        if self.is_border_cell((x,y)):
+            #return for the four corners
+            if x == 0:
+                if y == 0:
+                    #bottom left
+                    return set((Cell(1,0),Cell(0,1),Cell(1,1)))
+                elif y == (self.dim_y -1):
+                    #top left
+                    return set((Cell(0,self.dim_y-2),Cell(1,self.dim_y-1)))
+                else:
+                    #left border inside
+                    return set((Cell(0,y-1),Cell(0,y+1),Cell(1,y),Cell(1,y+1)))
+
+            elif x == self.dim_x - 1:
+                if y == 0:
+                    #bottom right
+                    return set((Cell(self.dim_x - 2,0),Cell(self.dim_x - 1,1)))
+                elif y == (self.dim_y -1):
+                    #top right
+                    return set((Cell(self.dim_x - 2,self.dim_y-2),
+                    Cell(self.dim_x - 2,self.dim_y-1),
+                    Cell(self.dim_x - 1,self.dim_y-2)))            
+        else:
+            return set((
+                Cell(x+1,y),
+                Cell(x-1,y),
+                Cell(x,y+1),
+                Cell(x+1,y+1),
+                Cell(x-1,y-1),
+                Cell(x,y-1))
+                )
+
+
+    def __repr__(self) -> str:
+        pass
+
 
 
 if __name__ == "__main__":
+    #TODO when happy these all work move to the test file
```

Comment:
Why not moving them there right away ? In a unit test or more ?

Reply:
I agree, can we do this in another pr? i think we can have another one where we add more, test,make it look nicer etc